pinot-query-planner/src/main/java/org/apache/calcite/jdbc/CalciteSchemaBuilder.java

```diff
@@ -61,4 +61,15 @@ public static CalciteSchema asRootSchema(Schema root) {
     }
     return rootSchema;
   }
+
+  public static CalciteSchema asSubSchema(Schema root, String name) {
+    CalciteSchema subSchema = CalciteSchema.createRootSchema(false, false, name, root);
+    SchemaPlus schemaPlus = subSchema.plus();
+    for (Map.Entry<String, List<Function>> e : FunctionRegistry.getRegisteredCalciteFunctionMap().entrySet()) {
```

Comment:
Do we actually need to add all functions to the subschema? I would expect that functions of the parent schema could be used and that would mean we only need to declare functions once in the root